### PR TITLE
add GraphQL documentation (except computation)

### PR DESCRIPTION
**Pull request by bombnp**
## Why did you create this PR
- There's no GraphQL doc at the moment.

## What did you do

- Added GraphQL documentation for Course, Override, Review, and User API
- Removed some unused GraphQL types
- Add some data validation to review service to match the docs

## Demo

[https://dev.cugetreg.com](https://dev.cugetreg.com)

## Checklist

- [ ] Deploy a demo
- [ ] Check browsers compatibility
- [ ] Wrote coverage tests

### CODE REVIEW

**Comment by bombnp** [comment]
Docs can be seen in GraphQL Playground (/_api/graphql) in the Docs drawer. Just go visit the queries and mutations and u can see the related types and enums

image

File: apps/cugetreg-api/src/review/review.service.ts [hunk 1]
```diff
@@ -32,6 +32,13 @@ export class ReviewService {
     { courseNo, semester, academicYear, studyProgram, rating, content }: CreateReviewInput,
     userId: string
   ): Promise<Review> {
+    if (rating) {
```
Comment (saengowp) [inline apps/cugetreg-api/src/review/review.service.ts:35]: I don't get this line. How does the bound check work?

File: apps/cugetreg-api/src/review/review.service.ts [hunk 2]
```diff
@@ -99,7 +106,20 @@ export class ReviewService {
     reviewInput: EditReviewInput,
     userId: string
   ): Promise<Review> {
+    if (reviewInput.rating) {
+      throw new BadRequestException({
+        reason: 'RATING_OUT_OF_BOUND',
+        message: `Rating must be between 0 and 10. Got ${reviewInput.rating}`,
```
Comment (saengowp) [inline apps/cugetreg-api/src/review/review.service.ts:112]: See above, What are the example output?

File: apps/cugetreg-api/src/user/user.graphql [hunk 3]
```diff
@@ -23,18 +22,36 @@ input CourseCartItemInput {
...
+
+  """
+  Returns current user's course cart.
+
+  Requires user authentication.
+  """
   courseCart: [CourseCartItem!]
-  calendarId: String
+
+  calendarId: String @deprecated(reason: "Will be redesigned.")
```
Comment (saengowp) [inline apps/cugetreg-api/src/user/user.graphql:48]: Is it unused?

**Review by saengowp — COMMENTED** [review]
Some question

File: apps/cugetreg-api/src/review/review.service.ts (hunk 1 above)
Comment (bombnp) [reply apps/cugetreg-api/src/review/review.service.ts:35]: I derped. This is supposed to be 'if (rating < 0 || rating > 10)'

Same as below

File: apps/cugetreg-api/src/user/user.graphql (hunk 3 above)
Comment (bombnp) [reply apps/cugetreg-api/src/user/user.graphql:48]: Im not sure if frontend currently accesses it or not. If not, then we can just remove.

**Review by saenyakorn — APPROVED** [review]
Other than @saengowp, LGTM

**Pull request merged**